### PR TITLE
Added user deletion procedure

### PR DESCRIPTION
**Pull request by petraselmer**
Should only be merged when https://github.com/neo4j/neo4j/pull/7297 is merged

### CODE REVIEW

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java [hunk 1]
```diff
@@ -406,6 +408,127 @@ public void roleManagement3() throws Exception
...
     }
 
+    //----------User deletion -----------
+
+    @Test
+    public void shouldDeleteUser() throws Exception
+    {
+        testCallEmpty( db, adminSubject, "CALL dbms.createUser('Craig', '1234', true)", null );
+        assertNotNull( "User Craig should exist", manager.getUser( "Craig" ) );
+        testCallEmpty( db, adminSubject, "CALL dbms.deleteUser('Craig')", null );
```
Comment (OliviaYtterbrink) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java]: I think this shouldn't fail silently but result in an error, just like creating a user that already exists.

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java [hunk 2]
```diff
@@ -406,6 +408,127 @@ public void roleManagement3() throws Exception
...
+    }
+
+    @Test
+    public void shouldAllowDeletingUserMultipleTimes() throws Exception
+    {
+        testCallEmpty( db, adminSubject, "CALL dbms.createUser('Craig', '1234', true)", null );
+        assertNotNull( "User Craig should exist", manager.getUser( "Craig" ) );
+        testCallEmpty( db, adminSubject, "CALL dbms.deleteUser('Craig')", null );
+        assertNull( "User Craig should not exist", manager.getUser( "Craig" ) );
+        testCallEmpty( db, adminSubject, "CALL dbms.deleteUser('Craig')", null );
```
Comment (OliviaYtterbrink) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java]: I think this should result in an error, just like creating a user that already exists.

Comment (OliviaYtterbrink) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java]: I think this should result in an error, just like creating a user that already exists.

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java [hunk 3]
```diff
@@ -406,6 +408,127 @@ public void roleManagement3() throws Exception
...
+        assertNull( "User Craig should not exist", manager.getUser( "Craig" ) );
+    }
+
+    @Test
+    public void shouldNotAllowNonAdminDeleteUser() throws Exception
+    {
+        testFailDeleteUser( readSubject );
+        testFailDeleteUser( writeSubject );
+        testFailDeleteUser( schemaSubject );
```
Comment (henriknyman) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java:434]: Maybe we should test with a "noneSubject" with no access rights for completion, and/or a subject that needs to change password?

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java (hunk 2 above)
Comment (henriknyman) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java]: I actually think it is nice if you do not fail unnecessarily on deletes

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java [hunk 4]
```diff
@@ -406,6 +414,136 @@ public void roleManagement3() throws Exception
...
+    public void shouldNotAllowNonAdminDeleteUser() throws Exception
+    {
+        testFailDeleteUser( noneSubject );
+        testFailDeleteUser( readSubject );
+        testFailDeleteUser( writeSubject );
+        testFailDeleteUser( schemaSubject );
+    }
+
+    @Test
+    public void shouldAllowDeletingUserMultipleTimes() throws Exception
```
Comment (fickludd) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java:438]: The name of the test should contain a 'Not' to match the behaviour tested

**Pull request merged**